app/controllers/login_controller.rb

```diff
@@ -23,9 +25,9 @@ def login
           save_unverified_user(user.id)
 
           if user.student?
```

Comment:
no need to check user role here? just `redirect_to(verify_email_by_pin_form_path)`